### PR TITLE
samples: nrf9160: modem_shell: ppp updates and link ipv6 MTU printing

### PR DESCRIPTION
**Pull request by jhirsi**
This PR adds bunch of improvements to PPP:
- ipv6 support enabled
- LTE link MTU to be informed to PC (via LCP MRU option)
- using semaphores for sockets instead of sleeping
- auto-restarting ppp net_if when dial up disconnected.
This enables reconnection possibility from PC side without
commanding manually ppp up/down from mosh.
- PPP net_if autostart disabled (needed by custom MTU)
- less UL buffering and not delaying ppp startup in bootup
- workarounding a modem failure for raw sckt sending (Jira: NCSDK-11858)

Additionally this add support for printing MTU for IPv6 within `link status` command.

Note: for full functionality this needs Zephyr changes from: https://github.com/nrfconnect/sdk-nrf/pull/6097

### CODE REVIEW

**Comment by peknis** [comment]
Would this change be worth an entry in the changelog?

**Review by trantanen — APPROVED** [review]

**Review by peknis — APPROVED** [review]

**Review by tokangas — APPROVED** [review]

File: samples/nrf9160/modem_shell/README.rst [hunk 1]
```diff
@@ -603,12 +603,12 @@ See :ref:`cmake_options` for more instructions on how to add these options.
...
 ===========
 
-To build the MoSh sample with PPP/dial up support for Windows, use the ``-DOVERLAY_CONFIG=overlay-ppp.conf`` option.
+To build the MoSh sample with PPP/dial up support, use the ``-DDTC_OVERLAY_FILE=ppp.overlay`` and ``-DOVERLAY_CONFIG=overlay-ppp.conf`` options.
 For example:
 
 .. code-block:: console
 
-   west build -p -b nrf9160dk_nrf9160_ns -d build -- -DOVERLAY_CONFIG=overlay-ppp.conf
+   west build -p -b nrf9160dk_nrf9160ns -- -DDTC_OVERLAY_FILE=ppp.overlay -DOVERLAY_CONFIG=overlay-ppp.conf
```
[the remark this answers is not in the document]
Comment (jhirsi) [reply samples/nrf9160/modem_shell/README.rst]: changed

File: samples/nrf9160/modem_shell/src/ppp/ppp_ctrl.c [hunk 2]
```diff
@@ -70,6 +150,34 @@ static void ppp_ctrl_link_default_pdn_status_handler(struct k_work *work_item)
...
+
+	mosh_print("PPP net_if: stopped");
+
+}
+
+static void ppp_ctrl_net_if_start(struct k_work *work_item)
+{
+	ARG_UNUSED(work_item);
+
+	bool done = false;
```
[the remark this answers is not in the document]
Comment (jhirsi) [reply samples/nrf9160/modem_shell/src/ppp/ppp_ctrl.c]: Yes, no need. Changed.

File: samples/nrf9160/modem_shell/overlay-ppp.conf [hunk 3]
```diff
@@ -97,9 +106,12 @@ CONFIG_NET_DEBUG_NET_PKT_ALLOC=y
 
 CONFIG_NET_CONTEXT_LOG_LEVEL_WRN=y
 CONFIG_UART_CONSOLE_LOG_LEVEL_WRN=y
-CONFIG_NET_BUF_LOG_LEVEL_WRN=y
-CONFIG_NET_BUF_LOG=y
-CONFIG_NET_BUF_WARN_ALLOC_INTERVAL=2
+
+CONFIG_NET_BUF_LOG_LEVEL_OFF=y
+CONFIG_NET_BUF_LOG=n
```
[the remark this answers is not in the document]
Comment (jhirsi) [reply samples/nrf9160/modem_shell/overlay-ppp.conf]: Yes, no need. Removed.

**Comment by jhirsi** [comment]
> Would this change be worth an entry in the changelog?

@peknis. I think it does, added one. Please, check.

File: doc/nrf/releases/release-notes-changelog.rst [hunk 4]
```diff
@@ -223,6 +223,7 @@ nRF9160 samples
   * Updated some samples to use DTS overlays instead of KConfig configuration files for setting up external flash memory.
   * Added support for nRF Cloud A-GPS and P-GPS.
     A-GPS support is enabled by default.
+  * PPP updates: IPv6 support, LTE link MTU to be informed to PC, autostarting of PPP improved and changes for enabling better performance.
```
Comment (peknis) [inline doc/nrf/releases/release-notes-changelog.rst:226]:
```suggestion
  * PPP updates: 
    * IPv6 support
    * LTE link MTU to be informed to PC
    * Improved autostart of PPP
    * Changes for better performance
```

**Review by peknis — CHANGES_REQUESTED** [review]

**Review by rlubos — APPROVED** [review]

**Review by peknis — APPROVED** [review]

**Pull request merged**